Microsoft.HealthVault.Fhir/Transformers/MedicationToHealthVault.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Get Real Health.  All rights reserved.
+// MIT License
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the ""Software""), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+
+using System.Linq;
+using Hl7.Fhir.Model;
+using Microsoft.HealthVault.Fhir.Codings;
+using Microsoft.HealthVault.ItemTypes;
+using Microsoft.HealthVault.Thing;
+using FhirMedication = Hl7.Fhir.Model.Medication;
+using HVMedication = Microsoft.HealthVault.ItemTypes.Medication;
+
+namespace Microsoft.HealthVault.Fhir.Transformers
+{
+    public static class MedicationToHealthVault
+    {
+        public static ThingBase ToHealthVault(this FhirMedication fhirMedication)
+        {
+            var hvMedication = new HVMedication();
+
+            hvMedication.Name = fhirMedication.Code.ToCodableValue();//.GetCodableValue();
+
+            var hasSingleIngredient = fhirMedication.Ingredient.Count == 1;
+            if (hasSingleIngredient)
```

Comment:
>hasSingleIngredient [](start = 16, length = 19)

I think there is a problem with this transformation. You cannot assume one ingredient is the generic name of the medication. We need to find a way to store in HealthVault the ingredients to preserve the information from FHIR.

As it is now, it really feels incomplete and limited as many medications contain more than one ingredient.

Reply:
This was done as a reverse operation of how we mapped `generic name` and `strength` to FHIR.

Am not very satisfied with this transformation and is open to better handling of this.

When transforming to FHIR,
>Current behavior is that `Generic name`, if any, is copied to as an `ingredient` and `strength` is added as its `amount`. If `generic name` is `null` and `strength` is `!null`, then name is copied as an `ingredient`.